webpack/assets/javascripts/foreman_tools.js

```diff
@@ -10,6 +10,39 @@ import { sprintf, translate as __ } from './react_app/common/I18n';
 
 import { showLoading, hideLoading } from './foreman_navigation';
 
+import store from './react_app/redux';
+import { openConfirmModal as coreOpenConfirmModal } from './react_app/components/ConfirmModal';
+
+export const openConfirmModal = options =>
+  store.dispatch(coreOpenConfirmModal(options));
+
+// override the jQuery UJS $.rails.allowAction
+$.rails.allowAction = function railsConfirmOverride(element) {
+  const message = element.data('confirm');
+  const answer = false;
+  let callback;
+  if (!message) return true;
+
+  if ($.rails.fire(element, 'confirm')) {
+    openConfirmModal({
+      title: __('Confirm'),
+      message,
+      onConfirm: () => {
+        callback = $.rails.fire(element, 'confirm:complete', [answer]);
+        if (callback) {
+          const oldAllowAction = $.rails.allowAction;
+          $.rails.allowAction = function allowAction() {
+            return true;
+          };
+          element.trigger('click');
+          $.rails.allowAction = oldAllowAction;
+        }
```

Comment:
`callback` will always be true here, we don't really have any callbacks on `confirm:complete` anywhere, certainly not ones that return `false`. we can simplify this to:
```suggestion
        const oldAllowAction = $.rails.allowAction;
        $.rails.allowAction = () => true;
        element.trigger('click');
        $.rails.allowAction = oldAllowAction;
```

Reply:
done, thanks!